changelog for storage common and blob (#974)
Azure/azure-sdk-for-cpp: sdk/storage/azure-storage-blobs/CHANGELOG.md

# Release History

> ## 12.0.0-beta.5 (2020-11-13)

Reviewer: @weshaggard would our build and release pipeline be able to handle this version jump from `1.0.0-beta.4` to `12.0.0-beta.5`?

### New Features

> * Support for replaceable HTTP transport layer.

Reviewer: Is this the new `TransportPolicyOptions` member as part of client options?

Author: yep

# Release History

## 12.0.0-beta.5 (2020-11-13)

### New Features

* Support for replaceable HTTP transport layer.

> * Add `version.hpp`.

Reviewer: super nit: If you would like to fix this going forward and for this current changelog, let's use past-tense similar to https://github.com/Azure/azure-sdk-for-cpp/pull/978
https://github.com/Azure/azure-sdk-for-c/releases/tag/1.0.0
https://github.com/Azure/azure-sdk-for-cpp/releases/tag/azure-core_1.0.0-beta.3

The writing style will continue to differ (and we don't really enforce this), so not a big deal either way, but we can try to make an effort on consistent styling. Up to you.

Reviewer 2: Actually, we did have this conversation within our team. It seems that it is just a personal style. If you take a look at [TensorFlow](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) you would find even within one file there are both styles. Search for `Add` will give you both past tense and simple present tense of the changelog.

Reviewer: Yes, I understand. Either is fine. Just wanted to mention it, in case we want to try to converge to one style. Leaving it as is, is fine as well :)

# Release History

## 12.0.0-beta.5 (2020-11-13)

### New Features

* Support for replaceable HTTP transport layer.
* Add `version.hpp`.

> ### Breaking Changes

Reviewer: I am surprised, but did we really have no bug fixes reported or made during this release? I don't see any myself, but maybe I missed it, so just want to confirm.

Author: no, i'm not really good at writing bugs.

* Move header `azure/storage/blobs/blob.hpp` to `azure/storage/blobs.hpp`.
* Service API return types which are typically suffixed with `Result` are moved to the `Models` sub-namespaces and everything else from the protocol layer is made private by moving to the `Details` namespace.
* Make XML serializer and deserializer private by moving them to the `Details` namespace.
* Remove `BlockBlobClientOptions`, `AppendBlobClientOptions` and `PageBlobClientOptions`, use `BlobClientOptions` instead.
* Rename `BlobSasBuilder::ToSasQueryParameters` to `BlobSasBuilder::GenerateSasToken`.

> * Make `SharedKeyCredential` a class.

Reviewer 2: Should this be in common instead?

# Release History

## 12.0.0-beta.5 (2020-11-13)

### New Features

* Support for replaceable HTTP transport layer.
* Add `version.hpp`.

### Breaking Changes

* Move header `azure/storage/blobs/blob.hpp` to `azure/storage/blobs.hpp`.
* Service API return types which are typically suffixed with `Result` are moved to the `Models` sub-namespaces and everything else from the protocol layer is made private by moving to the `Details` namespace.
* Make XML serializer and deserializer private by moving them to the `Details` namespace.
* Remove `BlockBlobClientOptions`, `AppendBlobClientOptions` and `PageBlobClientOptions`, use `BlobClientOptions` instead.
* Rename `BlobSasBuilder::ToSasQueryParameters` to `BlobSasBuilder::GenerateSasToken`.

### Other Changes and Improvements

* Default uploading/downloading concurrency is changed from 1 to 5.
* Remove support for specifying SAS version.

## 1.0.0-beta.4 (2020-10-16)

> ### Bug Fixes

Reviewer: bug fixes go below new features and breaking changes sections.

Author: is that a big deal?

Reviewer: Not a lot, but we do want to try to highlight the most important/impactful things here, and having similar section structure across the board helps that (also prevents subtle typos/issues where someone puts things in the wrong section accidentally).

See https://github.com/Azure/azure-sdk-for-cpp/pull/941#discussion_r521013885

# Release History

## 12.0.0-beta.5 (2020-11-13)

### New Features

> * Support for replaceable HTTP transport layer

Reviewer: ```suggestion
* Support for replaceable HTTP transport layer.
```

Author: no period for other sentences. we need to be consistent

Reviewer: I agree. I suggest we be consistent **the other way** and add the period (at least going forward).

Look at:
https://github.com/Azure/azure-sdk-for-cpp/releases/tag/azure-core_1.0.0-beta.3
https://github.com/Azure/azure-sdk-for-cpp/releases/tag/azure-storage-files-shares_1.0.0-beta.4
https://github.com/Azure/azure-sdk-for-cpp/releases/tag/azure-storage-blobs_1.0.0-beta.4
https://github.com/Azure/azure-sdk-for-c/releases/tag/1.1.0-beta.2

### Breaking Changes

> * API return types are moved to `Models` namespace, other stuff in protocol layer is moved to `Details` namespace

Reviewer: ```suggestion
* Service API return types which are typically suffixed with `Result` are moved to the `Models` sub-namespaces and everything else from the protocol layer is made private by moving to the `Details` namespace.
```

> * Move XML serializer and deserializer to `Details` namespace

Reviewer: ```suggestion
* Make XML serializer and deserializer private by moving them to the `Details` namespace.
```

> * Remove `BlockBlobClientOptions`, `AppendBlobClientOptions` and `PageBlobClientOptions`, use `BlobClientOptions` instead

Reviewer: ```suggestion
* Remove `BlockBlobClientOptions`, `AppendBlobClientOptions` and `PageBlobClientOptions`, use `BlobClientOptions` instead.
```

> * Rename `BlobSasBuilder::ToSasQueryParameters` to `BlobSasBuilder::GenerateSasToken`

Reviewer: nit: Here/elsewhere - end in a period.
```suggestion
* Rename `BlobSasBuilder::ToSasQueryParameters` to `BlobSasBuilder::GenerateSasToken`.
```

> ### Other Changes and Improvements

Reviewer: No bug fixes section?

I recommend going through the .hpp file changes from all commits between the last release and this one to make sure all user-facing/relevant changes are highlighted.

Author: I already did

> * Default uploading/downloading concurrency are changed to 5 from 1

Reviewer: ```suggestion
* Default uploading/downloading concurrency is changed from 1 to 5.
```

* Remove support for specifying SAS version

> ## 1.0.0-beta.5 (Unreleased)

Reviewer: Remove this section and take the contents and merge with above.

> ### New Features

Reviewer: Merge beta.3 changes into beta.4, similar to https://github.com/Azure/azure-sdk-for-cpp/pull/978#discussion_r522599423 to get consistency with the releases.

* CreateFromConnectionString now accepts unencoded blob name
* TagConditions is changed to nullable
* Variable name change: `Marker` is renamed to `ContinuationToken` for `ListContainersSegmentOptions`, `FindBlobsByTagsOptions` and `ListBlobsSegmentOptions`.

> * Varaible name change: `Marker` is renamed to `PreviousContinuationToken`, `NextMarker` is renamed to `ContinuationToken` for `FilterBlobsSegmentResult`, `ListContainersSegmentResult`, `ListBlobsByHierarchySegmentResult` and `ListBlobsFlatSegmentResult`.

Reviewer: Keep the period.

> * Varaible name change: `Marker` is renamed to `PreviousContinuationToken`, `NextMarker` is renamed to `ContinuationToken` for `FilterBlobsSegmentResult`, `ListContainersSegmentResult`, `ListBlobsByHierarchySegmentResult` and `ListBlobsFlatSegmentResult`

Reviewer: ```suggestion
* Variable name change: `Marker` is renamed to `PreviousContinuationToken`, `NextMarker` is renamed to `ContinuationToken` for `FilterBlobsSegmentResult`, `ListContainersSegmentResult`, `ListBlobsByHierarchySegmentResult` and `ListBlobsFlatSegmentResult`
```
